### PR TITLE
Improve default key management experience

### PR DESCRIPTION
**Pull request by trishankatdatadog**
- [x] Push `snapshot` key, just like `timestamp`, to Notary server.
- [x] Reuse `targets` key, just like `root`, across bundle repositories.
- ~[ ] Delegate all bundles from `targets to `targets/releases` for better security posture.~

### CODE REVIEW

File: pkg/tuf/keys.go
Comment (radu-matei) [inline pkg/tuf/keys.go:121]: Why exactly do we need to publish the changes first?

Comment (radu-matei) [reply pkg/tuf/keys.go:121]: Also, could we have a bit more verbose logging here?

**Review by radu-matei — APPROVED** [review]
This looks great!
The only comment I have is that this PR does diverge from the Docker CLI behaviour in a way _we_ agree is helpful.
However, there might be scenarios where users might actively want to have different pairs of `root` and `targets` keys for different bundles, scenario that is not covered by this PR.

I don't think we should hold up this PR because of it, but it would be fantastic if users / implementors could have this choice.

What do you think?


Also, before merging, could you please clean-up the commits so we have a clean commit history?
Thanks!

File: pkg/tuf/keys.go
Comment (trishankatdatadog) [reply pkg/tuf/keys.go:121]: Unfortunately, without first publishing, we are unable to list roles, IIRC from experience. Perhaps @justincormack could help to confirm, here?

Also, what do you mean by verbose logging? Where would you like to see them?

**Comment by trishankatdatadog** [comment]
> The only comment I have is that this PR does diverge from the Docker CLI behaviour in a way _we_ agree is helpful.
> However, there might be scenarios where users might actively want to have different pairs of `root` and `targets` keys for different bundles, scenario that is not covered by this PR.
> 
> I don't think we should hold up this PR because of it, but it would be fantastic if users / implementors could have this choice.

Agreed. I will file an issue to allow users to use different pairs of `root` and `targets` keys.

> Also, before merging, could you please clean-up the commits so we have a clean commit history?

Sorry, could you clarify this?

**Comment by radu-matei** [comment]
I mean that there are a few commits in this PR that, if they ended up in the main branch commit history, wouldn't add that much, but significantly pollute the overall history.

So ideally, you would squash some commits and only keep the logical commits that make sense. (And maybe rewrite their commit messages.)

(It is a sensible middle ground between squashing all commits into one, and keeping all of them),

**Comment by trishankatdatadog** [comment]
Squashed commits! Hopefully much cleaner.

**Comment by radu-matei** [comment]
This is excellent, thank you so much!

**Pull request merged**
